crd/apis/network/v1/network.go

```diff
@@ -18,6 +18,13 @@ func IsDefaultNetwork(networkName string) bool {
 	return networkName == DefaultNetworkName || networkName == DefaultPodNetworkName
 }
 
+// IsDefaultNetworkOrEmpty takes a network name and returns if it is a default network or empty.
+// Both DefaultNetworkName and DefaultPodNetworkName are considered as default network for compatibility purposes.
+// DefaultNetworkName will eventually be removed.
+func IsDefaultNetworkOrEmpty(networkName string) bool {
```

Comment:
if this is going to be read from a crd why is not defaulted on creation?
where is this function going to be used?

Reply:
@aojea 

>Where is this function going to be used?

The function is going to be used in our controllers and debug tooling wherever we need to determine whether a resource is associated with the default network or not.

>If this is going to be read from a crd why is not defaulted on creation? 

These CRDs existed before multi-networking so defaulting on creation wouldn't really work. I'm also hesitant to set a default since it seems that our default network has changed from one value to another.

We could update our controllers to add the default but even then, other tooling probably should handle the empty string case.

We don't have to merge this if it's not useful or desirable upstream. We have it as a helper function in our repo and I thought it might be useful.

Reply:
I feel is confusing, as you are describing it, this is a function that is workarounding a set of bad decisions we made in the past, I do not feel that we should expose it and newer versions should have the correct defaulting so users do not have to worry about empty vs old default vs new default vs ...

Reply:
They're not really bad decisions; they were just originally created before multi-networking was a thing. Both the API itself and the CR instances.

Even if multi-networking existed at the time, it's reasonable that a team might implement a CRD without multi-networking at first and then add it later. We want our tooling to still work in this case.

We shouldn't discount this scenario and should provide a reasonable story for it.

Reply:
bad worded, sorry, I try to mean that we should be better on handling breaking changes like these, all these changes on default names and empty values are breaking changes, and the CRD API version is still the same.
Kubernetes and CRDs apis allow to handle this by upgrading the versions , so you have v1alpha1, v1alpha2, v1beta1, ... and we must use them to handle this problems instead of this helpers functions, that are tied to this repository tags, so you can end with inconsistencies between the CRDs versions and this repo released version

Reply:
It isn't a breaking change though. The Network name field is optional and an empty string maps to the same behavior as before the field was added: using the default pod network.

It would be a breaking change if we required the Network field to have a specific value to keep the same behavior as before. In that case, I agree we would need a new CRD API version.

Reply:
> These CRDs existed before multi-networking so defaulting on creation wouldn't really work. I'm also hesitant to set a default since it seems that the default network name has changed from one value to another.

`it seems that the default network name has changed from one value to another.`  this is why I mean for breaking changes and not handling well the versioning of the APIs, I prefer not having these helpers and that the CRs are the source of truth and consistent